libs/payments/cart/src/lib/checkout.service.ts

```diff
@@ -426,6 +432,21 @@ export class CheckoutService {
         } else {
           throw new PaymentMethodUpdateFailedError(cart.id, customer.id);
         }
+      } else if (intent.status === 'requires_payment_method') {
+        const errorCode = isPaymentIntent(intent)
+          ? intent.last_payment_error?.code
+          : intent.last_setup_error?.code;
+        const declineCode = isPaymentIntent(intent)
+          ? intent.last_payment_error?.decline_code
+          : intent.last_setup_error?.decline_code;
+
+        throw handleIntentFailed(
```

Comment:
Can this be swapped over to the style of handling paypal errors? such as calling an equivalent of the `throwPaypalCodeError` method, which is responsible for throwing the errors internally rather than returning an error class
```suggestion
        throwIntentFailedError(
```